src/stock-items/add-stock-item/stock-item-rules/stock-rules.resource.ts

```diff
@@ -0,0 +1,79 @@
+import { openmrsFetch } from "@openmrs/esm-framework";
+import { ResourceFilterCriteria, toQueryParams } from "../../../core/api/api";
+import useSWR from "swr";
+import { PageableResult } from "../../../core/api/types/PageableResult";
+import { StockRule } from "../../../core/api/types/stockItem/StockRule";
+
+export interface StockSourceFilter extends ResourceFilterCriteria {
+  stockItemUuid?: string | null;
+}
+
+// Get Stock Rules
+export function useStockRules(filter: StockSourceFilter) {
+  const apiUrl = `ws/rest/v1/stockmanagement/stockrule${toQueryParams(filter)}`;
+
+  const { data, error, isLoading } = useSWR<
+    { data: PageableResult<StockRule> },
+    Error
+  >(apiUrl, openmrsFetch);
+
+  return {
+    items: data?.data || <PageableResult<StockRule>>{},
+    isLoading,
+    isError: error,
+  };
+}
+
+// Get Stock Rule
+export function useStockRule(id: string) {
+  const apiUrl = `ws/rest/v1/stockmanagement/stockrule/${id}`;
+  const { data, error, isLoading } = useSWR<{ data: StockRule }, Error>(
+    apiUrl,
+    openmrsFetch
+  );
+  return {
+    items: data.data ? data.data : [],
+    isLoading,
+    isError: error,
+  };
+}
+
+// Delete Stock Rule
+export function deleteStockRule(ids: string[]) {
+  let otherIds = ids.reduce((p, c, i) => {
+    if (i === 0) return p;
+    p += (p.length > 0 ? "," : "") + encodeURIComponent(c);
+    return p;
+  }, "");
+  if (otherIds.length > 0) {
+    otherIds = "?ids=" + otherIds;
+  }
+  const apiUrl = `ws/rest/v1/stockmanagement/stockrule/${ids[0]}${otherIds}`;
+  const abortController = new AbortController();
+  return openmrsFetch(apiUrl, {
+    method: "DELETE",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    signal: abortController.signal,
+  });
+}
```

Comment:
Please use readable variable names here. e.g 
```tsx 
deleteStockRule(stockRuleUuids: stringArray<string>)
 ```

hard to follow what `p c i` mean